data/StoreSelectionModel.js

```diff
@@ -35,29 +35,26 @@ export class StoreSelectionModel extends HoistModel {
         this.addReaction(this.cullSelectionReaction());
     }
 
-    /** @return {?Record} - single selected record, or null if multiple or no records selected. */
-    @computed
-    get singleRecord() {
-        const ids = this.ids;
-        return ids.length === 1 ? this.store.getById(ids[0]) : null;
-    }
-
     /** @return {Record[]} - currently selected Records. */
-    @computed
+    @computed({equals: comparer.shallow})
     get records() {
         return compact(this.ids.map(it => this.store.getById(it)));
     }
 
+    /** @return {?Record} - single selected record, or null if multiple or no records selected. */
+    get singleRecord() {
+        const {records} = this;
+        return records.length === 1 ? records[0] : null;
+    }
+
     /** @return {boolean} - true if selection is empty. */
-    @computed
     get isEmpty() {
-        return this.ids.length === 0;
+        return this.count === 0;
     }
 
     /** @return {number} - count of currently selected records. */
-    @computed
     get count() {
-        return this.ids.length;
+        return this.records.length;
```

Comment:
Maybe silly, but would this be a good place to use `@computed` given that the records can change without the count changing? Would that avoid possible re-renders of things that only wanted to depend on `count` or `isEmpty` directly?

Reply:
Yes!  If we think there are transitions where records goes from one collection to another with the same size, that would indeed cut down on noise.